src/ucp/proto/proto_multi.h

```diff
@@ -65,7 +95,7 @@ typedef struct {
     ucp_lane_index_t            num_lanes;    /* Number of lanes to use */
     size_t                      align_thresh; /* Cached value of threshold for
                                                  enabling data split alignment */
-    ucp_proto_multi_lane_priv_t lanes[0];     /* Array of lanes */
+    ucp_proto_multi_lane_priv_t lanes[UCP_MAX_LANES]; /* Array of lanes */
```

Comment:
Why changed to `UCP_MAX_LENGTH`? 

Minor: Align comments for other fields.

Reply:
I thought that rpriv is always `UCP_PROTO_PRIV_MAX` buffer allocated on the `ucp_proto_select _init_protocols` side. Maybe I am missing something, but I cannot find the place where we allocate `ucp_proto_multi_priv_t` using alloca.

Is that change made for some future improvements?

Reply:
For example here: https://github.com/openucx/ucx/pull/9695/files#diff-a430ba778d55dee29e9281cc476b4563ea1732a2462db82afb8dcd1c0bd1972aR249 we place ucp_proto_multi_priv_t  on the stack; if it had zero-size array of lanes we could not do that - we had to allocate with alloca and add space for the lanes (which create more code).

Reply:
Understood, thanks!